### ISSUE TITLE
Exception in DeduplicateTags. Sometimes. Only when no breakpoints are present.

### ISSUE THREAD

**Issue by fake-name** [starik222/BooruDatasetTagManager#80]
Ok, this is a weird issue I have been having issues tracking down:

![image](https://github.com/starik222/BooruDatasetTagManager/assets/1401239/3b8fb0d3-1b82-4bde-8b01-8533e6e40eee)

```
 	[External Code]	
>	BooruDatasetTagManager.dll!BooruDatasetTagManager.EditableTagList.DeduplicateTags() Line 214	C#
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.EditableTagList.ToString() Line 73	C#
 	[External Code]	
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.DatasetManager.DataItem.ToString() Line 402	C#
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.DatasetManager.DataItem.GetHashCode() Line 407	C#
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.DatasetManager.DataItem.GetTagsFromFile() Line 397	C#
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.DatasetManager.DataItem.DataItem(string imagePath, int imageSize) Line 370	C#
 	BooruDatasetTagManager.dll!BooruDatasetTagManager.DatasetManager.LoadFromFolder.AnonymousMethod__2(string x) Line 276	C#
 	[External Code]	

```

At the exception, `i` is 41, while `List` is 33 items long. 

I genuinely don't understand how this is happening. I removed the parallel load entirely (replacing it with a for loop in `DatasetManager.cs`) to eliminate any potential chance it's a concurrency issue (can that happen in C#?) and it still occurs. If I put a breakpoint in `DeduplicateTag()` to look at what things values are, and then manually continue execution every time the breakpoint is hit, the issue does not occur. 

I can patch around the issue by just sticking 

```
                if (i > List.Count - 1)
                    continue;
```
in the loop in `EditableTagList.cs`, but that is just patching over the issue.

**Comment by starik222** [comment]
This function runs synchronously, so there shouldn't be any concurrency issues. Can you describe the sequence of actions leading to the error? Or if the error appears on a specific dataset, send it to me. Using the latest version of the editor, I edited a large data set and have never encountered a similar problem.

**Comment by starik222** [comment]
Please try to work with the published release, if an exception also occurs in it, write, I will try to add a thread lock to this function.

**Comment by starik222** [comment]
Fixed in [2.0.1](https://github.com/starik222/BooruDatasetTagManager/releases/tag/v2.0.1)
